src/webgpu/web_platform/copyToTexture/image.spec.ts

```diff
@@ -146,6 +146,79 @@ g.test('from_image')
     );
   });
 
+g.test('from_fully_transparent_image')
+  .desc(
+    `
+  Test HTMLImageElement with alpha 0 can be copied to WebGPU texture correctly.
+
+  Use a prebaked 2x2 fully transparent image as source.
+
+  Then call copyExternalImageToTexture() to do a copy to the 0 mipLevel of dst texture,
+  and read the contents out to compare with the HTMLImageElement contents.
+  When dest alpha mode is:
+  - premultiplied, the content should be (0, 0, 0, 0)
+  - not premultiplied, the content should be the same as prebaked
+    pixel values (255, 102, 153, 0).
+
+  The tests covers:
+  - Source HTMLImageElement is fully transparent with valid dest alphaMode.
+  And the expected results are all passed.
+  `
+  )
+  .params(u => u.combine('dstPremultiplied', [true, false]))
+  .beforeAllSubcases(t => {
+    if (typeof HTMLImageElement === 'undefined') t.skip('HTMLImageElement not available');
+  })
+  .fn(async t => {
+    const { dstPremultiplied } = t.params;
+
+    const kColorFormat = 'rgba8unorm';
+    const kImageWidth = 2;
+    const kImageHeight = 2;
+
+    const imageCanvas = document.createElement('canvas');
+    imageCanvas.width = kImageWidth;
+    imageCanvas.height = kImageHeight;
+
+    // Prebaked fully transparent image with content (255, 102, 153, 0)
+    const image = new Image(kImageWidth, kImageHeight);
```

Comment:
We should probably test this for other inputs type other than `Image` (`HTMLImageElement`) as well. Not sure if the Chromium bug was specific to HTMLImageElement, but it's useful coverage anyway.

Could you file a bug for it?

(Maybe we can find a way to factor these tests so we don't have to duplicate it too much. For example this test could easily have a parameter for whether we go through HTMLImageElement or ImageBitmap, or could even be generalized to many input types, since the input data is trivial.)

Reply:
>Not sure if the Chromium bug was specific to HTMLImageElement, but it's useful coverage anyway.

I think the chromium bug is specific to HTMLImageElement. The root cause for that bug is the decoder decodes image with `premultiplied-alpha` and `CopyEI2T` loses the info. So we need a redecoded path for this. I think other inputs (maybe `HTMLVideoElement` ?) doesn't have such problem?

Reply:
Still, it would probably be relatively easy for a bug to creep in the future, or in another browser's implementation, so more coverage would be useful.

Reply:
(Please feel free to land this PR though.)

Reply:
Thanks and I've got time to back to this PR recently.  Let me file an issue for other inputs in case we don't forget it.

Reply:
Thanks! Can you link to the issue?

Reply:
Actually #3350 is still open so I'll just write this down there (if you haven't already filed another issue)